tools/mo/openvino/tools/mo/front/tf/placeholder_ext.py

```diff
@@ -13,9 +16,31 @@ class PlaceholderFrontExtractor(FrontExtractorOp):
 
     @classmethod
     def extract(cls, node):
+        shape = shape_array([])
+        # Extract output shape from `shape` attribute
+        extracted_shape = tf_tensor_shape(node.pb.attr["shape"].shape)
+        if len(extracted_shape) != 0 and not node.pb.attr["shape"].shape.unknown_rank:
```

Comment:
let us use `_output_shape` iff. `node.pb.attr["shape"].shape.unknown_rank is True`, otherwise, just use `shape`
Also, if you use `_output_shape` please check its presence.

Reply:
>let us use _output_shape iff. node.pb.attr["shape"].shape.unknown_rank is True
I am not agree, we have cases in our validation where `shape` is empty and `shape.unknown_rank` is not provided and it set to False while model parsing. So if we will use this check we will not be able to do convert these models.

Reply:
>Also, if you use _output_shape please check its presence.

It already done, in line 28:
`if len(extracted_output_shapes) == 1: # check if attribute not empty`

Reply:
what if `_output_shape` attrivute does not exist in `node.pb.attr` at all. I shared example above.

Reply:
Please check my answers in comments above and below.